### ISSUE TITLE
cppcoreguidelines-avoid-const-or-ref-data-members - false positive

### ISSUE THREAD

**Issue by johngladp** [llvm/llvm-project#64011]
With clang-tidy v16 the following false positive error is generated:
```
   struct extended_OVERLAPPED_t
   {      
      OVERLAPPED overlapped{};
      packet_descriptor_IOCP_t* const outer;

      extended_OVERLAPPED_t(
         packet_descriptor_IOCP_t& outer_ )
         : outer( &outer_ )
      {
      }
   };
```

error G417914D0: member 'outer' of type 'packet_descriptor_IOCP_t *const' is const qualified [cppcoreguidelines-avoid-const-or-ref-data-members,-warnings-as-errors]

The pointer "outer" is not const therefore this error is incorrect.  In this instance the value pointed to by outer is const which is not a violation of the uncopyable/immovable C++ core guideline (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref) this rule is enforcing.
